src/main/java/org/ohdsi/circe/check/checkers/ConceptSetCriteriaCheck.java

```diff
@@ -63,7 +63,8 @@ protected void checkCriteria(Criteria criteria, String groupName, WarningReporte
                         .then(addWarning))
                 .isA(ConditionOccurrence.class)
                 .then(c -> match(((ConditionOccurrence)c))
-                        .when(conditionOccurrence -> Objects.isNull(conditionOccurrence.codesetId))
+                        .when(conditionOccurrence -> Objects.isNull(conditionOccurrence.codesetId)
+                                && Objects.isNull(conditionOccurrence.conditionSourceConcept))
```

Comment:
Additionally check:
DeviceExposure
DrugExposure
Measurement
Observation
ProcedureOccurrence
Specimen
VisitOccurrence

Reply:
I added check for SourceConcept for your list